Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -2066,3 +2080,26 @@ extension StatusTableViewController: ServicesViewModelDelegate {
     }
 
 }
+
+extension StatusTableViewController: GlucoseUnitPublisher {
+    func addGlucoseUnitObserver(_ observer: GlucoseUnitObserver, queue: DispatchQueue = .main) {
+        glucoseUnitObservers.insert(observer, queue: queue)
+    }
+
+    func removeGlucoseUnitObserver(_ observer: GlucoseUnitObserver) {
+        glucoseUnitObservers.removeElement(observer)
+    }
+
+    func notifyObserversOfGlucoseUnitChange(to glucoseUnit: HKUnit) {
+        glucoseUnitObservers.forEach { $0.glucoseUnitDidChange(to: glucoseUnit) }
+    }
+
+    private func publishGlucoseUnitToCGMManagerIfNeeded() {
```

Comment:
[nit] Couldn't we just make the `CGMManager` protocol also conform to the `GlucoseUnitObserver` protocol and go from there? Seems cleaner.

Reply:
My initial thought was that not all CGM managers may want to observe the glucose unit change. However, I'll have a go at it.